### PR TITLE
Add performance test running image-classifier

### PR DESCRIPTION
**Pull request by virtan**
Summary:
Image-classifier gets warmup argument to warm up before running performance measuring cycles.

Reviewed By: hl475

Differential Revision: D17505452

### CODE REVIEW

File: tools/loader/ImageClassifier.cpp [hunk 1]
```diff
@@ -671,12 +677,12 @@ int main(int argc, char **argv) {
...
-      runBenchmark(name, loader, std::move(contexts), requestCount);
+      runBenchmark(name, loader, std::move(contexts), requestCount, warmup);
       if (timeOpt) {
         timer.stopTimer();
-        llvm::outs() << llvm::formatv("Wall time per item (s): {0:f4}\n",
-                                      timer.getTotalTime().getWallTime() /
-                                          iterationsOpt);
+        llvm::outs() << llvm::formatv(
+            "Wall time per item (s): {0:f4}\n",
+            timer.getTotalTime().getWallTime() / iterationsOpt + warmup);
```
Comment (rdzhabarov) [inline tools/loader/ImageClassifier.cpp]: this does not seem correct: timer.getTotalTime().getWallTime() / iterationsOpt + warmup => timer.getTotalTime().getWallTime() / (iterationsOpt + warmup)

File: tools/loader/ImageClassifier.cpp [hunk 2]
```diff
@@ -464,10 +470,10 @@ static void runInference(runtime::HostManager *hostManager, std::string name,
 static void
 runBenchmark(std::string name, Loader &loader,
              std::vector<std::unique_ptr<ExecutionContext>> contexts,
-             unsigned requestCount) {
+             unsigned requestCount, unsigned warmUp) {
```
Comment (rdzhabarov) [inline tools/loader/ImageClassifier.cpp]: update doxygen to reflect new param

File: tools/loader/ImageClassifier.cpp [hunk 3]
```diff
@@ -433,14 +439,14 @@ static void runInference(runtime::HostManager *hostManager, std::string name,
                          std::unique_ptr<ExecutionContext> batch,
                          std::promise<void> &runPromise,
                          std::atomic<unsigned> &inflight,
-                         std::atomic<int> &dispatched) {
+                         std::atomic<int> &dispatched, unsigned warmUp) {
```
Comment (rdzhabarov) [inline tools/loader/ImageClassifier.cpp]: ditto

**Comment by rdzhabarov** [comment]
Looking great. Can you update the PR title (since here it's just about ImageClassifier changes).

File: tools/loader/ImageClassifier.cpp (hunk 1 above)
Comment (virtan) [reply tools/loader/ImageClassifier.cpp]: Uff, thanks for catching it

File: tools/loader/ImageClassifier.cpp [hunk 4]
```diff
@@ -450,24 +456,26 @@ static void runInference(runtime::HostManager *hostManager, std::string name,
...
         if (--inflight == 0) {
           runPromise.set_value();
         }
       });
 }
 
-/// Run the requested number of benchmark requests \p requestCount through the
-/// HostManager from the \p loader using the provided context pool \p contexts
+/// Run the requested number of benchmark requests \p requestCount prepended by
+/// \p warmUp cycles
```
Comment (rdzhabarov) [inline tools/loader/ImageClassifier.cpp]: this looks broken, can you prettify this

**Pull request closed without merging**